src/oslogin_sshca.cc

```diff
@@ -203,12 +203,17 @@ static int GetExtension(const char *key, size_t k_len, char **exts) {
     goto out;
   }
 
-  // Skip valid principals.
-  if (GetString(&key_b64, &n_len, NULL, NULL) < 0) {
+  // Extract valid principals.
+  if (GetString(&key_b64, &n_len, &tmp_prin, &tmp_prin_len) < 0) {
     SysLogErr("Failed to skip cert's \"valid principals\" field.");
     goto out;
   }
 
+  if (GetString(&tmp_prin, &tmp_prin_len, principal, NULL) < 0) {
+    SysLogErr("Failed to read princiapl.");
```

Comment:
Typo spotted. ;)

But also, what is this function call doing? It appears to be dissecting the `valid_principals` field, but it could use a comment explaining why it's valid to call this function with only 3 of 4 parameters set (previously it was only called with all 4, or with 2 of 4).